RubeesEat/Model/EqualityComparer/BillPropertyEqualityComparer.cs

```diff
@@ -0,0 +1,34 @@
+using System.Diagnostics.CodeAnalysis;
+
+namespace RubeesEat.Model.EqualityComparer;
+
+public class BillPropertyEqualityComparer : IEqualityComparer<Bill>
+{
+    private EntryLinePropertyEqualityComparer _entryLineComparer = new();
+    public bool Equals(Bill? x, Bill? y)
+    {
+        if (ReferenceEquals(x, y))
+            return true;
+        if (x == null || y == null)
+            return false;
+        return x.Id == y.Id &&
+               x.Date == y.Date &&
+               x.Description == y.Description &&
+               x.EntryLines.SequenceEqual(y.EntryLines, _entryLineComparer);
+    }
+
+    public int GetHashCode([DisallowNull] Bill bill)
```

Comment:
Seems like the DisallowNull doesn't do anything here

Reply:
It doesn't here. It's for static code analysis tools to know that no null should be entered. It tells the compiler that, so it can be useful if we ever accidentally enter a null value in our code